crates/egui/src/widgets/slider.rs

```diff
@@ -578,23 +578,23 @@ impl<'a> Slider<'a> {
                 self.value_from_position(position, position_range.clone())
             };
             if !self.snap_values.is_empty() {
+                // We use ui points as the unit for the measurements:
                 let mut closest_snap = 0.0;
-                let norm_value = normalized_from_value(new_value, self.range.clone(), &self.spec);
-                let mut closest_distance = f64::INFINITY;
-                for snap in self.snap_values.iter().copied() {
-                    let snap_norm_value =
-                        normalized_from_value(snap, self.range.clone(), &self.spec);
-                    let distance = (norm_value - snap_norm_value).abs();
+                let new_value_pos = self.position_from_value(new_value, position_range.clone());
+                let mut closest_distance = f32::INFINITY;
+                for snap_value in self.snap_values.iter().copied() {
+                    let snap_pos = self.position_from_value(snap_value, position_range.clone());
+                    let distance = (new_value_pos - snap_pos).abs();
                     if distance < closest_distance {
-                        closest_snap = snap;
+                        closest_snap = snap_value;
                         closest_distance = distance;
                     } else {
                         break;
                     }
                 }
 
-                // Divide aim_radius by the granularity we want
-                if self.snap_values_only || closest_distance < ui.input().aim_radius() as f64 / 15.0 {
+                // Multiply aim_radius by the granularity we want
+                if self.snap_values_only || closest_distance < ui.input(|i| i.aim_radius()) * 5.0 {
```

Comment:
I think some kind of multiplier is necessary here, but I'm not sure how we should determine what it is. 

With no multiplier, the aim radius is too small for it to actually snap. The precision is too tight for using a mouse.
![2c9d976390](https://user-images.githubusercontent.com/19912588/217844295-7b79ed53-62a7-4299-a29a-9b306267e761.gif)

By multiplying the aim radius by 5 as I have here, the snap works more like I expect.
![c0c8b43de6](https://user-images.githubusercontent.com/19912588/217844805-e9399558-975e-4fae-827e-b181b4fd0866.gif)